Sming/Wiring/WHashMap.h

```diff
@@ -199,86 +218,85 @@ class HashMap
     ||
     || @return true if it is contained in this HashMap
     */
-    bool contains(K key) const
+    bool contains(const K& key) const
+    {
+      return indexOf(key) >= 0;
+    }
+
+    /*
+     || @description
+     || | Remove entry at given index
+     || #
+     ||
+     || @parameter index location to remove from this HashMap
+     */
+    void removeAt(unsigned index)
     {
-      for (int i = 0; i < currentIndex; i++)
+      if (index >= currentIndex)
+        return;
+
+      for (unsigned i = index + 1; i < size; i++)
       {
-        if (cb_comparator)
-        {
-          if (cb_comparator(key, *keys[i]))
-          {
-            return true;
-          }
-        }
-        else
-        {
-          if (key == *keys[i])
-          {
-            return true;
-          }
-        }
+        *keys[i - 1] = *keys[i];
```

Comment:
Can you try the following test code with your new modifications?

```
	HttpHeaders headers;
	headers["X-Auth-Header"] = "Value Data";
	headers["x-Auth-header"] = "Value Data New";

	debugf("Header: %s", headers["X-AUTH-HEADER"].c_str());

	headers.remove("x-auth-header");

	debugf("Headers count: %d", headers.count());

```

Expected result:
```
Header: Value Data New
Headers count: 0
```

Reply:
That is weird. The comparator doesn't get set... not entirely sure why but I'm happier having a default constructor anyway. Thanks for spotting that one!

Reply:
Looks better now.

> I think the compiler decided to call the HashMap constructor with the default of nullptr instead of the inherited default constructor. Compiler bug? Or am I missing something?

My guess is that it has nothing to do with the compiler but a confirmation will require disassembling of the compiled binary and analyzing  it. Anyway you can try reverting the constructor code back to `HashMap(comparator compare=0)` instead of ``HashMap(comparator compare=nullptr)  +  comparator cb_comparator = nullptr` and check if it works.

Reply:
> I think the compiler decided to call the HashMap constructor with the default of nullptr instead of the inherited default constructor. 

That can be checked easier with a debugger.